File: Lab7/Question8/src/Dog.java
public class Dog extends Animal{

    public Dog(String name) {
        super(name);
    }

    @Override
    public int getLegs() {
        return 4;
    }

    @Override
    public boolean canFly() {
        return false;
    }

    public static void testDog() {
        Dog dog = new Dog("Inu");
        System.out.println(dog.getName() == "Inu");
        System.out.println(dog.canFly() == false);
        System.out.println(dog.getLegs() == 4);
    }
}

File: Lab7/Dog.kt
class Dog : Animal {
    constructor(name: String) : super(name)

    override fun getLegs(): Int {
        return 4
    }

    override fun canFly(): Boolean {
        return false
    }

    companion object {
        fun testDog() {
            val dog = Dog("Inu")
            println(dog.getName() == "Inu")
            println(dog.canFly() == false)
            println(dog.getLegs() == 4)
        }
    }
}